pkg/controller/clusters/status.go

```diff
@@ -110,6 +122,70 @@ func updateControlPlaneInitializedCondition(cluster *anywherev1.Cluster, kcp *co
 	conditions.MarkTrue(cluster, anywherev1.ControlPlaneInitializedCondition)
 }
 
+// updateWorkersReadyCondition updates the WorkersReadyConditon condition after checking the state of the worker node groups
+// in the cluster.
+func updateWorkersReadyCondition(cluster *anywherev1.Cluster, machineDeployments []clusterv1.MachineDeployment) {
+	initializedCondition := conditions.Get(cluster, anywherev1.ControlPlaneInitializedCondition)
+	if initializedCondition.Status != "True" {
+		conditions.MarkFalse(cluster, anywherev1.WorkersReadyConditon, initializedCondition.Reason, initializedCondition.Severity, initializedCondition.Message)
+		return
+	}
+
+	// TODO: Here we should update the status based on the DefaultCNIConfigured condition. When it's false and skipUpgrades
+	// is not enabled for the CNI, WorkersReady should be marked false with the corresponding reason from the
+	// DefaultCNIConfigured condition.
```

Comment:
I don't necessarily agree with this
This makes an assumption about the order in which conditions are set and the order the clusters are reconciled
Why not just set the reason based on the observed state of the cluster?

Reply:
I had been assume an order the clusters are reconciled here, but yeah if we don't want that we shouldn't do this. It's also probably not needed at all, since the workers condition should focus on the state of the workers. I'll remove this comment